### PR TITLE
fix(parser): force comma between EXPLAIN options

### LINKED ISSUE #5878: parser: EXPLAIN options without comma is accepted
### Describe the bug

_No response_

### To Reproduce

`explain (verbose  trace) select 1;`

### Expected behavior

syntax error
should only accept `explain (verbose, trace) select 1;`

### Additional context

BTW as mentioned earlier, `explain () select 1` is also accepted, but rejected by pg. This is a minor problem.

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title. Also bans empty `EXPLAIN` option in accordance with pg.

```
dev=> explain (trace, ) select 1;
ERROR:  QueryError: sql parser error: Unrecognized EXPLAIN option ")"

dev=> explain (trace, ver) select 1;
ERROR:  QueryError: sql parser error: Unrecognized EXPLAIN option "ver"

dev=> explain (trace  verbose) select 1;
ERROR:  QueryError: sql parser error: Expected ), found: verbose

dev=> explain () select 1;
ERROR:  QueryError: sql parser error: Unrecognized EXPLAIN option ")"
```

## Checklist

- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

Closes #5878

### CODE REVIEW

File: src/sqlparser/src/parser.rs [hunk 1]
```diff
@@ -1324,6 +1324,23 @@ impl Parser {
         }
     }
 
+    /// Call `f` with the current token and returns the result if the current token is not one of
+    /// the expected keywords, or consume it if it is
+    pub fn expect_one_of_keywords_or<F>(
```
Comment (xxchan) [inline src/sqlparser/src/parser.rs:1329]: Why not simply `expect_one_of_keywords`?

**Review by xxchan — COMMENTED** [review]
Add a test case?

File: src/sqlparser/src/parser.rs (hunk 1 above)
Comment (Gun9niR) [reply src/sqlparser/src/parser.rs:1329]: To customize the error message

Comment (xxchan) [reply src/sqlparser/src/parser.rs:1329]: I feel it not very necessary and has inconsistent style. Maybe just simply don't customize error message, or 
```
expect_one_of_keywords(..).map_err(|e| match e {
    ParserError::TokenizerError(_) => unreachable!(),
    ParserError::ParserError(e) => {
        ParserError::ParserError(format!("Unrecognized EXPLAIN option: {e}"))
    }
})?;
```

**Review by xxchan — APPROVED** [review]

**Pull request merged**